addons/sms/tools/sms_api.py

```diff
@@ -0,0 +1,74 @@
+        account = self.env['iap.account'].get('sms')
+        params['account_token'] = account.account_token
+        endpoint = self.env['ir.config_parameter'].sudo().get_param('sms.endpoint', DEFAULT_ENDPOINT)
+        return iap_tools.iap_jsonrpc(endpoint + local_endpoint, params=params, timeout=timeout)
+
+    def _send_sms_batch(self, messages, delivery_reports_url=False):
+        """ Send SMS using IAP in batch mode
+
+        :param list messages: list of SMS (grouped by content) to send:
+          formatted as ```[
+              {
+                  'content' : str,
+                  'numbers' : [
+                      { 'uuid' : str, 'number' : str },
+                      { 'uuid' : str, 'number' : str },
+                      ...
+                  ]
+              }, ...
+          ]```
+        :param str delivery_reports_url: db base_url to receive delivery reports
+        :return: response from the endpoint called, which is a list of results
+          formatted as ```[
+              {
+                  uuid: UUID of the request,
+                  state: ONE of: {
+                      'success', 'processing', 'server_error', 'unregistered', 'insufficient_credit',
+                      'wrong_number_format', 'duplicate_message', 'country_not_supported', 'registration_needed',
+                  },
+                  credit: Optional: Credits spent to send SMS (provided if the actual price is known)
+              }, ...
+          ]```
+        """
+        params = {
+            'messages': messages,
+            'webhook_url': delivery_reports_url and werkzeug.urls.url_join(delivery_reports_url, '/sms/status')
+        }
+        return self._contact_iap('/iap/sms/3/send', params)
+
+    def _get_sms_api_error_messages(self):
+        """Return a mapping of `_send_sms_batch` errors to an error message.
+
+        We prefer a dict instead of a message-per-error-state based method so that we only call
+        'get_credits_url' once and avoid extra RPC calls."""
+
+        buy_credits_url = self.env.sudo()['iap.account'].get_credits_url(service_name='sms')
+        buy_credits = '<a href="%s" target="_blank">%s</a>' % (
+            buy_credits_url,
+            _('Buy credits.')
+        )
+        return {
+            'unregistered': _("You don't have an eligible IAP account."),
+            'insufficient_credit': ' '.join([_("You don't have enough credits on your IAP account."), buy_credits]),
+            'wrong_number_format': _("The number you're trying to reach is not correctly formatted."),
+            'duplicate_message': _("This SMS has been removed as the number was already used."),
+            'country_not_supported': _("The destination country is not supported."),
+            'incompatible_content': _("The content of the message violates rules applied by our providers."),
+            'registration_needed': _("Country-specific registration required."),
```

Comment:
Hello @flch-odoo ,

This is indeed a new error status that has been introduced in the version 3 of the `send` API.

To help the user solve the error, we should add a link to the country registration page, that would be :
`https://sms.api.odoo.com/api/sms/1/account?account_token=$SMS_ACCOUNT_TOKEN`
where `$SMS_ACCOUNT_TOKEN` can be obtained by :
```py
self.env['iap.account'].sudo().get('sms').account_token
```